### PR TITLE
Developing RESTful APIs with Loopback

### PR DESCRIPTION
**Pull request by unicodeveloper**

### CODE REVIEW

File: _posts/2017-08-22-loopback.markdown [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+![Tables and Data](https://cdn.auth0.com/blog/loopback/tables.png)
+
+## Test the API
+
+Now that our API is connected to MySQL. Let's test our API for persistence. Make a `GET` request to `http://localhost:3000/api/planets`.
+
+![Get Planets](https://cdn.auth0.com/blog/loopback/planets.png)
+_Initial GET Request_
+
+Go ahead and make a `POST` request to the URL.
```
Comment (brunokrebs) [inline _posts/2017-08-22-loopback.markdown]: I prefer so much `curl` commands... The reader can just copy and paste to execute the command. But well, that is just my preference.

File: _posts/2017-08-22-loopback.markdown [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+
+## Secure the Star Wars API
+
+The majority of the APIs we use on a daily basis have a means of authorizing users to make changes to them. We'll go ahead and secure some of these API endpoints with JSON Web Tokens.
+
+*JSON Web Tokens*, commonly known as JWTs, are tokens that are used to authenticate users on applications. This technology has gained popularity over the past few years because it enables backends to accept requests simply by validating the contents of these JWTs. That is, applications that use JWTs no longer have to hold cookies or other session data about their users. This characteristic facilitates scalability while keeping applications secure.
+
+Whenever the user wants to access a protected route or resource (an endpoint), the user agent must send the JWT, usually in the `Authorization` header using the [*Bearer schema*](http://self-issued.info/docs/draft-ietf-oauth-v2-bearer.html), along with the request.
+
+The API receives a request with a JWT, the first thing it does is to validate the token. This consists of a series of steps, and if any of these fails then, the request must be rejected. The following list shows the validation steps needed:
```
Comment (brunokrebs) [inline _posts/2017-08-22-loopback.markdown]: "The API receives" => "When the API receives"

File: _posts/2017-08-22-loopback.markdown [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+
+For starters, if you haven't done so yet, this is a good time to sign up for a [free Auth0 account](javascript:signup\(\)). Having an Auth0 account, the first thing that we must do is to [create a new API on the dashboard](https://manage.auth0.com/#/apis). An API is an entity that represents an external resource, capable of accepting and responding to protected resource requests made by clients. And we are dealing with an API here, SWAPI (Star Wars API).
+
+> [Auth0 offers a generous **free tier**](https://auth0.com/pricing) to get started with modern authentication.
+
+Login to your Auth0 [management dashboard](https://manage.auth0.com) and create a new API client. If you don't already have the APIs menu item, you can enable it by going to your [Account Settings](https://manage.auth0.com/#/account/advanced) and in the **Advanced** tab, scroll down until you see **Enable APIs Section** and flip the switch.
```
Comment (brunokrebs) [inline _posts/2017-08-22-loopback.markdown]: This doesn't apply anymore. The APIs item is shown by default.

File: _posts/2017-08-22-loopback.markdown [hunk 4]
```diff
@@ -0,0 +1,606 @@
+---
```
Comment (brunokrebs) [inline _posts/2017-08-22-loopback.markdown]: Oh, I forgot to mention, you have to add two to three tweet buttons in the article.

File: _posts/2017-08-22-loopback.markdown [hunk 5]
```diff
@@ -0,0 +1,606 @@
...
+layout: post
+title: "Developing RESTful APIs with Loopback"
+description: Learn how to build and secure RESTful APIs with Loopback.
+date: 2017-08-24 8:30
+category: Technical Guide, Backend, Loopback
+author:
+  name: Prosper Otemuyiwa
+  url: https://twitter.com/unicodeveloper?lang=en
```
[the remark this answers is not in the document]
Comment (unicodeveloper) [reply _posts/2017-08-22-loopback.markdown]: lol, that's what I have been using

File: _posts/2017-08-22-loopback.markdown [hunk 6]
```diff
@@ -0,0 +1,606 @@
...
+Let's go with the second option. It's more effective. And we need test data. We'll write an automigration script to programmatically create the tables.
+
+Create a new file, `server/boot/create-data-tables.js` and add code to it:
+
+```js
+module.exports = function(app) {
+  app.dataSources.mysqlDs.automigrate('People', function(err) {
+    if (err) throw err;
+
+    app.models.People.create([{
```
[the remark this answers is not in the document]
Comment (unicodeveloper) [reply _posts/2017-08-22-loopback.markdown]: Omitted on purpose

File: _posts/2017-08-22-loopback.markdown [hunk 7]
```diff
@@ -0,0 +1,606 @@
...
+};
+```
+
+The bunch of code above simply create the tables and seeds them with the data provided. Whenever your app is initialized, the script will run and ensure the tables exist and are seeded with the right data.
+
+Head over to your console and start your app again:
+
+```bash
+node .
+```
```
[the remark this answers is not in the document]
Comment (unicodeveloper) [reply _posts/2017-08-22-loopback.markdown]: The script expects singular. Loopback will automatically transform it into the right plurals.

File: _posts/2017-08-22-loopback.markdown (hunk 6 above)
[the remark this answers is not in the document]
Comment (brunokrebs) [reply _posts/2017-08-22-loopback.markdown]: Error is thrown due to this omission.

[the remark this answers is not in the document]
Comment (unicodeveloper) [reply _posts/2017-08-22-loopback.markdown]: Oh I see what you mean

**Pull request merged**